### DISCUSSION TITLE
My Link does not work on detail page

### DISCUSSION THREAD

**Discussion by liebeckm** [neuefische/questions#563] [Web]
Hey @ all,

I am working on building a booking confirmation page right now. I already have a overview page with all offers and each offer also has a detail page. Now, I want to add a "book" button on the detail page. The button is already implemented on the main page (means the overview page) but it does not work on the details page. The link is always with "undefined" in the end.
http://localhost:3000/detail-ride/booking-confirmation/undefined

I suspect, that I have to change the code from the details page into something similar like the List component but since I don't need to map, I don't know how to do this...?
I think I just miss something here. What could be a good solution on this?

- [This](https://github.com/liebeckm/capstone-project/blob/booking-confirmation/pages/index.js) is the Overview page, which has the "buchen" link implemented and is working
- [This](https://github.com/liebeckm/capstone-project/blob/booking-confirmation/components/List/List.js) is the List component which is implemented in said overview page
- [This](https://github.com/liebeckm/capstone-project/blob/booking-confirmation/pages/detail-ride/%5Bslug%5D.js) is the details page, where I also implemented the link but it's not working
- [This](https://github.com/liebeckm/capstone-project/blob/booking-confirmation/pages/booking-confirmation/%5Bslug%5D.js) is the booking confirmation page

Thank you! 🫶

**Comment by actyralf** [comment]
You are trying to read the `slug` property from `rides` which is an array.
I guess you have to use the property from the object you are displaying (`extractedRide`).

Could look like this:

```js
const {
    slug
    driver,
    car,
    date,
    time,
    departure,
    meetingpoint,
    destination,
    seats,
    price,
    currency,
  } = extractedRide;
```
and
```js
<StyledLink href={`./booking-confirmation/${slug}`}>
     Buchen
</StyledLink>
```

**Reply by liebeckm** [reply]
Hey @actyralf thank you for the quick suggestion! I changed the Link as you told me to but couldn't add slug to the const, because it got defined earlier in the code. Then, I took the folder"booking-confirmation" into the detail-ride folder. I am not sure if this is the cleanest solution, but now it works 😌

**Reply by actyralf** [reply]
Ah - yes, I didn't see that `slug` was already defined. Yes, it should work then with just using `slug` in the `Link` component. I did not get what you did with the folders, but now the `slug` part of the URL should no longer be undefined.
